File: leetcode/476/476-a.cpp
class Solution {
public:
    int findComplement(int num) {
        int ans = 0;
        int bit = 0;

        while(num != 0){
            ans |= ((num & 1) ^ 1) << bit;
            num >>= 1;
            bit++;
        }
        return ans;
    }
};

File: leetcode/476/476-a.rs
impl Solution {
    pub fn find_complement(mut num: i32) -> i32 {
        let mut ans = 0;
        let mut bit = 0;

        while (num != 0) {
            ans |= ((num & 1) ^ 1) << bit;
            num >>= 1;
            bit += 1;
        }
        ans
    }
}
